Userland/Libraries/LibJS/AST.cpp

```diff
@@ -792,6 +792,8 @@ void NullLiteral::dump(int indent) const
 bool BindingPattern::contains_expression() const
 {
     for (auto& entry : entries) {
+        if (entry.name.has<NonnullRefPtr<Expression const>>())
+            return true;
```

Comment:
This is not testable in any way?

Reply:
This change makes sure that following:
"a. NOTE: A separate Environment Record is needed to ensure that bindings created by direct eval calls in the formal parameter list are outside the environment where parameters are declared."

is actually executed for following test262 test https://github.com/tc39/test262/blob/main/test/language/function-code/eval-param-env-with-computed-key.js

but this test also passes if it doesn't.